Add dpdk-app-centos docker image files to repo.

Billy99 opened the pull request:
Update main README to account for recent updates to repo.

Signed-off-by: Billy McFall <22157057+Billy99@users.noreply.github.com>

Billy99 committed:
Add dpdk-app-centos docker image files to repo.
Update main README to account for recent updates to repo.

Signed-off-by: Billy McFall <22157057+Billy99@users.noreply.github.com>

zenghui-shi commented on Dockerfile:
```diff
@@ -13,7 +13,7 @@ RUN rpm --import https://mirror.go-repo.io/centos/RPM-GPG-KEY-GO-REPO && \
     make
 
-RUN cp /usr/src/app-netutil/bin/samples /usr/bin/app
+RUN cp /usr/src/app-netutil/bin/go_app /usr/bin/app
```
Might need to apply the same change to `Dockerfile.rhel7` which is for openshift build.

zenghui-shi replied:
Shall we move the `Dockerfile` and `deployments/pod.yaml` to sample dir? for example, into `c_sample` or/and `go_sample`. This way, it will be consistent with dpdk samples. And we probably could remove `Dockerfile.rhel7` as we will not be delivering an OpenShift net-util images officially.

zenghui-shi commented on README.md:
```diff
@@ -1,41 +1,164 @@
+#### DPDK Sample Image
+The initial problem `app-netutil` is trying to solve is to collect initial
+configuration data for a DPDK application running in a container. The DPDK
+Library is written is C, so there is a sample Docker Image that leverages
```
`is written is C` => `in written in C`.

Billy99 replied:
ack

zenghui-shi commented on samples/dpdk_app/sriov/README.md:
```diff
@@ -0,0 +1,161 @@
+#  SR-IOV Deployment with dpdk-app-centos image
```
I didn't find info on this dpdk-sriov sample about when and how to bind VF with vfio-pci driver, do we already have it somewhere else?

Billy99 replied:
ACK - Updated the " SR-IOV Setup" section to indicate this setup assumes VFs have already been created. There is already a link to both the SR-IOV CNI and Device Plugin in this section.

zenghui-shi commented on samples/dpdk_app/dpdk-app-centos/README.md:
```diff
@@ -0,0 +1,201 @@
+## SR-IOV CNI and SR-IOV Device Plugin
+The SR-IOV Device Plugin detects and tracks the VFs associated with an
+SR-IOV PF. The SR-IOV CNI updates the VF as needed and the PCI Address
+associated woth the VF is passed tp the container via environmental
```
`woth` => `with`;  `tp` => `to`

Billy99 replied:
ACK

zenghui-shi commented on samples/dpdk_app/dpdk-app-centos/README.md:
````diff
@@ -0,0 +1,201 @@
+$ l3fwd -n 4 -l 1 --master-lcore 1 -w 0000:01:0a.6 -w 0000:01:02.1 -- -p 0x3 -P --config="(0,0,1),(1,0,1)" --parse-ptype
+```
+
+This Docker image is tweaking this a little. Before `l3fwd` is built, the
````
Thanks Billy for writing this sample! I cannot believe that it's changing the source code with sed automatically. That should have required a mount of work :+1:

jianzzha replied:
agree, truly nice work!

zenghui-shi commented on samples/dpdk_app/sriov/README.md:
```diff
@@ -0,0 +1,161 @@
+to the each VF using VLANs. 
+
+## Download the sample yaml files
+Use the following steps to download the sample YAML files:
```
This is very good example using upstream templates for SR-IOV deployment!
I assume later we may want to have a similar example using openshift or a pointer to openshift doc for SR-IOV deployment.

zshi-redhat reviewed (commented):
@Billy99 Thanks for working on the examples and document!
It looks great overall!
Just some minor comments inline.

Billy99 committed:
Address merge comments.
Fixed typos. Moved the 'testpod' files under the samples directory.
Cleaned up a few of the README files.

Signed-off-by: Billy McFall <22157057+Billy99@users.noreply.github.com>

jianzzha commented on samples/dpdk_app/sriov/sriov-pod-1.yaml:
```diff
@@ -0,0 +1,44 @@
+      items:
+        - path: "labels"
+          fieldRef:
+            fieldPath: metadata.labels
```
labels  not defined in this example pod metadata, for clarity it might be better to remove it here.

jianzzha replied:
by the way, I thought you wanted to grep pci address info from env, but env not defined for this pod ...

Billy99 replied:
The "labels" will be added by Admission Controller as part of the downward API. I haven't been using Admission Controller up to this point, so I manually added. I can add a comment to the documentation at some later point highlighting that.

Device Plugin adds the PCI Address dynamically. I didn't think I needed to do anything here.

jianzzha replied:
good to know. I used env for my setup to pass in the pci info and I would love to get rid of that.

jianzzha replied:
I just had a chance to test out the env passed in by the device plugin and I saw two possibilities:
1) PCIDEVICE_INTEL_COM_INTEL_SRIOV_EAST=0000:86:0a.7
    PCIDEVICE_INTEL_COM_INTEL_SRIOV_WEST=0000:86:02.7
2) PCIDEVICE_SRIOV_COM_MLNX_SRIOV_NETDEVICE=0000:af:00.1,0000:af:00.2

One of them actually failed my script. Just want to check if you have both forms taken care.

Billy99 replied:
If you are referring to app-netutil handling the environmental variables, Zenghui coded that and I just used it. I have not tested with Mellanox, I don't have any cards. So would not surprise me if there is an issue or two with Mellanox. I should be getting a Mellanox card this week for vDPA testing, I can see if I can get SR-IOV up and running and test with it.

jianzzha commented:
This is great work. Do you plan to add one more pod example for ovs-dpdk? or annotations use for that is already documented somewhere? thanks

Billy99 commented:
> This is great work. Do you plan to add one more pod example for ovs-dpdk? or annotations use for that is already documented somewhere? thanks

On the fence. The same Docker Image files are already in Userspace CNI (https://github.com/intel/userspace-cni-network-plugin) along with the associated YAML files for both OvS and for VPP. Because it is a complex setup (need OvS or VPP running on host), wasn't sure how much to duplicate here.

zenghui-shi commented:
/lgtm
@Billy99 thanks!
